images/base/files/usr/local/bin/entrypoint

```diff
@@ -62,32 +62,34 @@ fix_mount() {
   mount --make-rshared /
 }
 
+# helper used by fix_cgroup
 mount_kubelet_cgroup_root() {
-  cgroup_root=$1
-  subsystem=$2
-
-  if [ -n "${cgroup_root}" ]; then
-    # This is because we set Kubelet's cgroup-root to `/kubelet` by
-    # default. We have to do that because otherwise, it'll collide
-    # with the cgroups used by the Kubelet running on the host if we
-    # run kind cluster within a Kubernetes pod, resulting in random
-    # processes to be killed.
-    mkdir -p "${subsystem}/${cgroup_root}"
-    if [ "${subsystem}" == "/sys/fs/cgroup/cpuset" ]; then
-      # This is needed. Otherwise, assigning process to the cgroup
-      # (or any nested cgroup) would result in ENOSPC.
-      cat "${subsystem}/cpuset.cpus" > "${subsystem}/${cgroup_root}/cpuset.cpus"
-      cat "${subsystem}/cpuset.mems" > "${subsystem}/${cgroup_root}/cpuset.mems"
-    fi
-    # We need to perform a self bind mount here because otherwise,
-    # systemd might delete the cgroup unintentionally before the
-    # kubelet starts.
-    mount --bind "${subsystem}/${cgroup_root}" "${subsystem}/${cgroup_root}"
+  local cgroup_root=$1
+  local subsystem=$2
+  if [ -z "${cgroup_root}" ]; then
+    return 0
+  fi
+  mkdir -p "${subsystem}/${cgroup_root}"
+  if [ "${subsystem}" == "/sys/fs/cgroup/cpuset" ]; then
+    # This is needed. Otherwise, assigning process to the cgroup
+    # (or any nested cgroup) would result in ENOSPC.
+    cat "${subsystem}/cpuset.cpus" > "${subsystem}/${cgroup_root}/cpuset.cpus"
+    cat "${subsystem}/cpuset.mems" > "${subsystem}/${cgroup_root}/cpuset.mems"
   fi
+  # We need to perform a self bind mount here because otherwise,
+  # systemd might delete the cgroup unintentionally before the
+  # kubelet starts.
+  mount --bind "${subsystem}/${cgroup_root}" "${subsystem}/${cgroup_root}"
 }
 
 fix_cgroup() {
   echo 'INFO: fix cgroup mounts for all subsystems'
+  # see: https://d2iq.com/blog/running-kind-inside-a-kubernetes-cluster-for-continuous-integration
+  # capture initial state before 
+  local current_cgroup
+  current_cgroup=$(grep systemd /proc/self/cgroup | cut -d: -f3)
+  local cgroup_subsystems
+  cgroup_subsystems=$(findmnt -lun -o source,target -t cgroup | grep "${current_cgroup}" | awk '{print $2}')
```

Comment:
this is failing because this path is missing in github actions
```
Dec 16 08:12:51 kind-control-plane kubelet[3263]: I1216 08:12:51.213645    3263 cgroup_manager_linux.go:294] The Cgroup [kubelet] has some missing paths: [/sys/fs/cgroup/systemd/kubelet/kubelet]
```

Reply:
kubelet error says is missing that path :thinking:

Reply:
these are the paths that the kubelet is building

> Dec 16 09:02:53 kind-control-plane kubelet[22444]: I1216 09:02:53.236466   22444 cgroup_manager_linux.go:269] DEBUG Cgroups [kubelet] paths map[blkio:/sys/fs/cgroup/blkio/kubelet cpu:/sys/fs/cgroup/
cpu,cpuacct/kubelet cpuacct:/sys/fs/cgroup/cpu,cpuacct/kubelet cpuset:/sys/fs/cgroup/cpuset/kubelet devices:/sys/fs/cgroup/devices/kubelet freezer:/sys/fs/cgroup/freezer/kubelet hugetlb:/sys/fs/cgro
up/hugetlb/kubelet memory:/sys/fs/cgroup/memory/kubelet net_cls:/sys/fs/cgroup/net_cls,net_prio/kubelet net_prio:/sys/fs/cgroup/net_cls,net_prio/kubelet perf_event:/sys/fs/cgroup/perf_event/kubelet 
pids:/sys/fs/cgroup/pids/kubelet rdma:/sys/fs/cgroup/rdma/kubelet systemd:/sys/fs/cgroup/systemd/kubelet/kubelet]

Reply:
why is building the systemd path incorrectly the kubelet?
> systemd:/sys/fs/cgroup/systemd/kubelet/kubelet]